db/batch/batch_impl.go

```diff
@@ -24,12 +24,13 @@ type (
 
 	// cachedBatch implements the CachedBatch interface
 	cachedBatch struct {
-		lock sync.RWMutex
-		KVStoreCache
+		lock         sync.RWMutex
 		kvStoreBatch *baseKVStoreBatch
 		tag          int            // latest snapshot + 1
 		batchShots   []int          // snapshots of batch are merely size of write queue at time of snapshot
-		cacheShots   []KVStoreCache // snapshots of cache
+		caches       []KVStoreCache // snapshots of cache
+		keyTags      map[hash.Hash160][]int
```

Comment:
IMO, it would be easier for contributors to understand if some comments could be added about the design of `keyTags` and `tagKeys`